docker-compose/docker-compose.multiserver.mariadb.yml

```diff
@@ -62,8 +73,15 @@ services:
       - ACTIVEMQ_BROKER_PASSWORD=password
     ports:
       - "8080:8080"
+    depends_on:
+      - mariadb
+      - activemq
```

Comment:
While this will ensure that the Opencast containers are only started after MariaDB and ActiveMQ, it will not ensure that those two are ready for Opencast to connect to. The Opencast image has a mechanism to wait for the DB connection and Opencast itself waits for ActiveMQ. This is therefore unnecessary and has not the desired effect anyway.

Reply:
Even so, the containers have built their processes to wait for the database and the message broker, this helps to declutter the Logfile and ensures good practice.

Reply:
I disagree on the "good practice" part for the following reasons:

1. This does not solve the dependency problem as it does not guarantee that the downstream services are ready (see <https://docs.docker.com/v17.09/compose/startup-order/>).
1. Docker Swarm [completely ignores this setting](https://docs.docker.com/v17.09/compose/compose-file/#depends_on). In these environments the idea is that your application is resilient enough to handle temporary connection losses and arbitrary startup orders.
1. Each Opencast image therefore first checks if the DB is available with the help of a [little Java program](https://github.com/opencast/opencast-docker/blob/a8fe650006b5926edbb718954a8c6e9240955136/Dockerfiles/allinone/assets/scripts/TryToConnectToDb.java) and only then continues to start Opencast. For ActiveMQ this is handled directly by Opencast. This should be the fastest way to boot up the cluster.

I'm therefore not a fan of the `depends_on` option and I'm not recommending using it.

I can see that it at least documents the dependencies and maybe it leads to decluttered logs during startup (you can also get the logs for each service with `docker-compose logs -f <myservice>`). But do you think this is enough of a justification? As this is an example, I'm afraid people will get a false understanding of `depends_on`.

Reply:
Meanwhile, I support that the application itself has to do their own checks, with the sources you give me now I fell betrayed by Docker  😫. The "depends_on" statement loses all their meaning if not can check if a container is ready.

I will delete that lines